lib/models/addon.js

```diff
@@ -699,6 +702,24 @@ let addonProto = {
     @return {Tree} App file tree
   */
   treeForApp(tree) {
+    if (!experiments.MODULE_UNIFICATION || this.project.isModuleUnification()) {
```

Comment:
pretty sure this should be `|| !this.project.isModuleUnification()) ` with a not (!)

Reply:
discussed in slack, I think we actually need the following guard:

```js
if (!experiments.MODULE_UNIFICATION) {
  return tree;
} else if (project.isModuleUnification() && this.isModuleUnification()) {
  return null;
}
```

And add a new `isModuleUnification` method to the addon base class...